pilot/pkg/networking/core/cluster_waypoint.go

```diff
@@ -312,18 +327,62 @@ func (cb *ClusterBuilder) buildWaypointInboundVIP(proxy *model.Proxy, svcs map[h
 	return clusters
 }
 
+// buildWaypointInnerConnectOriginate creates a cluster that innitiate inner HBONE tunnel of double-HBONE.
+// InnerConnectOrigiante cluster is responsible for creating the inner CONNECT tunnel and forwarding to an internal listener that
+// will wrap the traffic into outer HBONE tunnel.
+func (cb *ClusterBuilder) buildWaypointInnerConnectOriginate(proxy *model.Proxy, push *model.PushContext) *cluster.Cluster {
+	// Normally for clusters that just redirect to internal listeners we would use util.DefaultInternalUpstreamTransportSocket.
+	// util.DefaultInternalUpstreamTransportSocket is a InternalUpstreamTransport socket wrapping a RawBufferTransport socket.
+	// For double HBONE we want something slightly different - we still need to use InternalUpstreamTransport socket because
+	// we redirect to an internal listener, but instead of keeping data as-is we want to encrypt it using TLS, so instead of the
+	// RawBufferTransport we want to wrap a TLS transport socket.
+	tlsCtx := buildCommonConnectTLSContext(proxy, push)
+	sec_model.EnforceCompliance(tlsCtx)
+	transportSocket := util.InternalUpstreamTransportSocket("internal_upstream_with_tls", &core.TransportSocket{
+		Name: "tls",
+		ConfigType: &core.TransportSocket_TypedConfig{TypedConfig: protoconv.MessageToAny(&tlsv3.UpstreamTlsContext{
+			CommonTlsContext: tlsCtx,
+		})},
+	})
+
+	c := &cluster.Cluster{
+		Name:                 InnerConnectOriginate,
```

Comment:
nit: do we want to put double-hbone in the cluster name to add a bit more context?

Reply:
Yeah the constant name